### DISCUSSION TITLE
Config question

### DISCUSSION THREAD

**Discussion by mfinegold-cambia** [renovatebot/renovate#9689] [Request Help]
Our gitlab pipeline uses a shell rather than docker executor so can't use the image tag in my gitlab ci yaml ( so looks like I can't use the other Whitesource renovate-runner project).  I'd like to create a new separate gitlab project that pulls in the renovate docker image and include the renovate config in my project.  Currently I'm including/exporting an env file of vars like RENOVATE_REPOSITORIES, RENOVATE_INCLUDE_PATHS etc in the container, but there are so many renovate config params that using env vars for this config seems cumbersome.  

I'd rather use a json or js file for config (passing it in as RENOVATE_CONFIG_FILE or something )  but seems like this approach won't work unless it's in the target repo?  Is there a way I can pass that json/js file to renovate if the file is not in the target repo (CI_PROJECT_DIR)?

**Comment by viceice** [comment]
you can run renovate from npm package, then you only need to add a `config.js` or `config.json` to the runner repo root and run renovate from there.

if you have a docker daemon available, you can set `binarySource=docker`  so renovate will use docker sidecar container to run other binary tools.

Please provide your pipeline, yo we can help you to workout best practices.

**Reply by mfinegold-cambia** [reply]
Thanks for the reply.  I'm not able to install renovate on our company shared gitlab runner or put files there since that is not owned by me.  So, am looking for a way to pass either RENOVATE_CONFIG_FILE or a file of equivalent renovate env variables to renovate in a docker run.  

Current approach is to create a project ( renovate-repo ) with a Dockerfile that pulls from the renovate docker image.  Then am doing a docker build and docker run of it in my pipeline like the below.  The renovate-config file referenced below is a file of env vars like RENOVATE_ENDPOINT, RENOVATE_PLATFORM etc.  But using env variables is cumbersome, especially for things like RENOVATE_PACKAGE_RULES which is a array/object, so would rather pass a config.js ( but RENOVATE_CONFIG_FILE  doesn't work in this scenario it seems ).  How do I pass a config.js to renovate in the docker run here?

scan-dependencies:
  stage: dependencies
  only:
    refs:
      - schedules
    variables:
      - $JOB == "RENOVATE-SCAN"
  script:
    - docker build -f ./renovate-repo/Dockerfile -t chp-cps-renovate  . 
    - docker run --env-file ./renovate-repo/config/renovate-config chp-cps-renovate
